docs/develop/guides/compact-javascript-runtime.mdx

```diff
@@ -0,0 +1,441 @@
+
+const expectedRuntimeVersionString = '0.8.1';
+
+const expectedRuntimeVersion = expectedRuntimeVersionString.split('-')[0].split('.').map(Number);
+
+const actualRuntimeVersion = __compactRuntime.versionString.split('-')[0].split('.').map(Number);
+
+if (expectedRuntimeVersion[0] != actualRuntimeVersion[0] || [...])
+
+  throw new __compactRuntime.CompactError(`Version mismatch...`);
+
+{ 
+
+  const MAX_FIELD = 52435875175126190479447740508185965837690552500527637822603658699938581184512n;
+
+  if (__compactRuntime.MAX_FIELD !== MAX_FIELD)
+
+    throw new __compactRuntime.CompactError(`compiler thinks maximum field value is ${MAX_FIELD}...`);
+
+}
+```
+
+This boilerplate ensures that your locally installed runtime matches what the Compact compiler expected when it generated the circuits.
+
+### **2\. Type Definitions and Descriptors**
+
+Next, the file defines **enumerations** and **type descriptors**.
+
+These tell the runtime how to encode and decode the data types used in your contract — like integers, strings, or custom structs.
+
+```typescript 
+var State;
+
+(function (State) {
+
+  State[State['VACANT'] = 0] = 'VACANT';
+
+  State[State['OCCUPIED'] = 1] = 'OCCUPIED';
+
+})(State = exports.State || (exports.State = {}));
+
+const _descriptor_1 = new __compactRuntime.CompactTypeUnsignedInteger(18446744073709551615n, 8);
+
+const _descriptor_2 = new __compactRuntime.CompactTypeBytes(32);
+
+const _descriptor_3 = new __compactRuntime.CompactTypeBoolean();
+
+const _descriptor_4 = new __compactRuntime.CompactTypeOpaqueString();
+```
+
+Each `_descriptor_*` object defines how values of a given type are represented in zero-knowledge proofs, ensuring consistent serialization between JavaScript and the underlying circuits.
+
+### **3\. Composite Types and Data Structures**
+
+Complex Compact types, like `Option`, `Maybe`, or structured records, are represented as small JavaScript classes that combine primitive descriptors.
+
+```typescript
+class _Maybe_0 {
+
+  alignment() {
+
+    return _descriptor_3.alignment().concat(_descriptor_4.alignment());
+
+  }
+
+  fromValue(value_0) {
+
+    return {
+
+      is_some: _descriptor_3.fromValue(value_0),
+
+      value: _descriptor_4.fromValue(value_0)
+
+    };
+
+  }
+
+  toValue(value_0) {
+
+    return _descriptor_3.toValue(value_0.is_some)
+
+      .concat(_descriptor_4.toValue(value_0.value));
+
+  }
+
+}
+```
+
+Each of these classes provides methods like `fromValue()` and `toValue()` to convert between **JavaScript objects** and **Compact-compatible encodings**.
+
+### **4\. The Contract Class and Circuit Wrappers**
+
+This is where things get interesting; the generated runtime defines a `Contract` class that mirrors your Compact contract’s **entry points** (such as `post`, `takeDown`, etc.).
+
+These methods don’t execute business logic directly; instead, they prepare inputs, call the pseudo-zero-knowledge circuits, and handle proof data.
```

Comment:
I've got know idea what a pseudo-zero-knowledge circuit is.  I don't think it's a real thing.